client/src/components/search/searchBar/search-bar.tsx

```diff
@@ -211,9 +214,9 @@ export class SearchBar extends Component<SearchBarProps, SearchBarState> {
                     }}
                     showLoadingIndicator={false}
                     translations={{
-                      submitTitle: t('icons.magnifier'),
-                      resetTitle: t('icons.input-reset'),
-                      placeholder: t('search.placeholder')
+                      submitTitle: submitTitle,
+                      resetTitle: resetTitle,
+                      placeholder: placeholder
```

Comment:
```suggestion
                      submitTitle: t('icons.magnifier'),
                      resetTitle: t('icons.input-reset'),
                      placeholder: t('search.placeholder')r
```

Reply:
The type of `translations` prop of `SearchBox` is `{ [key: string]: string | ((...args: any[]) => any); } | undefined`, and `t`  function's return type is `DefaultTFuncReturn` (or, `string | null`), so to avoid typescript error, can we go these lines as shown below?

```
    submitTitle: t<string>('icons.magnifier'),
    resetTitle: t<string>('icons.input-reset'),
    placeholder: t<string>('search.placeholder')
```

Reply:
I think it will be something like `submitTitle: t<string, string>('icons.magnifier'),`, but my issue with it is that the type isn't true, so whenever we need to add a similar change, we will need to add this type for every change instead of removing `preval` and importing the other language files without the need for it, which should fix the core of the issue with the type.

I prefer to keep, your current implementation, over overwriting the types, because I have done the same, so when we need to remove `preval` there will be one scenario to sort instead of two.